runtime/decimal.c

```diff
@@ -107,6 +109,28 @@ TaggedPtrPanicCode finish(decQuad *dq, decContext *cx) {
     return result;
 }
 
+TaggedPtrPanicCode _bal_decimal_from_float(double val) {
+    TaggedPtrPanicCode result;
+    if (isnan(val) || isinf(val)) {
+        result.panicCode = PANIC_INVALID_CONVERSION;
```

Comment:
We don't need a separate error here.  nan should produce PANIC_INVALID_DECIMAL; inf should produce PANIC_ARITHMETIC_OVERFLOW.

Reply:
Fixed